### PR TITLE
5 get articles

### PR DESCRIPTION
**Pull request by FreemindNation**
Hi, please look at this endpoint and kindly provide some feedback

- add tests for GET /api/articles
- add controller
- add model
- add error handlers
- add middlewares

### CODE REVIEW

File: __tests__/api.test.js [hunk 1]
```diff
@@ -95,10 +95,47 @@ describe('GET /api/articles/:article_id', () => {
...
+                    article_img_url: expect.any(String),
+                    comment_count: expect.any(Number)
+                });
+            });
+
+            expect(articles).toBeSortedBy('created_at',{ descending: true, });
+        });
+    });
+
+    test('404: responds with "Route not found" if passed an invalid route', () => {
```
Comment (nc-stephen-corke) [inline __tests__/api.test.js:130]: Great 404 test - it doesn't belong to this endpoint though so I'd pop it in it's own describe block

File: api/models/articles-models.js [hunk 2]
```diff
@@ -5,8 +5,39 @@ exports.selectArticleById = (article_id)=> {
...
+  GROUP BY 
+  articles.author, 
+  articles.title, 
+  articles.article_id, 
+  articles.topic, 
+  articles.created_at, 
+  articles.votes, 
+  articles.article_img_url
+  ORDER BY articles.created_at DESC
+  `
```
Comment (nc-stephen-corke) [inline api/models/articles-models.js:36]: The GROUP BY seems a little verbose and I wonder whether it is working as expected. We want to count the comments that have the same article_id so you can probably just GROUP BY that

**Review by nc-stephen-corke — COMMENTED** [review]
Looks good but have a little look at your SQL when you get a sec :)

**Comment by FreemindNation** [comment]
Thank you. changes will be made in the following branch

**Pull request merged**